prometheus/reporter.go

```diff
@@ -291,6 +291,18 @@ func NewReporter(opts Options) Reporter {
 	}
 	if opts.OnRegisterError == nil {
 		opts.OnRegisterError = func(err error) {
+			// n.b. Because our forked Prometheus client does not actually emit
```

Comment:
this seems to maybe warrant its own PR. lets not overload this diff with different concerns. Even though its to allow us to test i think since this is changing core behavior it should be its own diff.

Reply:
I agree, but behavior isn't changing here - just the error that's being panicked. Users who provide their own `OnRegisterError` will see no difference as a result of this change.

The only difference that updating our Prom fork would do is allowing us to type switch on the error, and that comes at an unknown (but potentially high) cost of upgrading dependents' use of the Prom client, which I don't think is quite as valuable here. They're two distinct concerns, but one does not necessarily follow from the other.

Reply:
(The same holds for would-be recoverers of this panic - this is a builtin error, so they will receive the same underlying error type via any `recover()` calls, modulo the additional (arguably more helpful) error message.

Reply:
ok, seems your case is reasonable.

Reply:
I'd still vote for this to go in a separate PR, given it's not related to mutex changes. It seems like a very independent change (definitely a good change, just not related to mutexes)